ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -2279,7 +2279,10 @@ public synchronized void write() throws StoreException {
             toWrite.writeIndexSegmentToFile(toWrite.getEndOffset());
             toWrite.seal();
           }
-          currentInfo.writeIndexSegmentToFile(indexEndOffsetBeforeFlush);
+          //Compaction may seal the current last index segment.
+          if (!currentInfo.isSealed()) {
+            currentInfo.writeIndexSegmentToFile(indexEndOffsetBeforeFlush);
```

Comment:
Since compaction may seal it, this is not thread safe: compaction many seal it between 2282 and 2283.

Reply:
Updated.